File: leetcode/00007_reverse-integer/7-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        num = 0
        digit = 0
        absX = abs(x)
        while absX:
            digit = absX % 10
            absX = absX // 10
            num = num * 10 + digit

        limit = 2**31 - 1
        if num > limit:
            return 0

        if x < 0:
            return num * -1
        else:
            return num

File: leetcode/00007_reverse-integer/7-reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let num = 0,
        digit = 0;
    let absX = Math.abs(x);
    while (absX) {
        digit = absX % 10;
        absX = parseInt(absX / 10);
        num = num * 10 + digit;
    }
    let limit = Math.pow(2, 31) - 1;
    if (num > limit) {
        return 0;
    }
    return x < 0 ? num * -1 : num;
};